core-bundle/src/Picker/AbstractContentPickerProvider.php

```diff
@@ -0,0 +1,139 @@
+<?php
+
+declare(strict_types=1);
+
+/*
+ * This file is part of Contao.
+ *
+ * (c) Leo Feyer
+ *
+ * @license LGPL-3.0-or-later
+ */
+
+namespace Contao\CoreBundle\Picker;
+
+use Doctrine\DBAL\Connection;
+use Knp\Menu\FactoryInterface;
+use Symfony\Component\Routing\RouterInterface;
+use Symfony\Component\Security\Core\Security;
+use Symfony\Component\Translation\TranslatorInterface;
+
+abstract class AbstractContentPickerProvider extends AbstractPickerProvider implements DcaPickerProviderInterface
+{
+    /**
+     * @var Security
+     */
+    private $security;
+
+    /**
+     * @var Connection
+     */
+    private $connection;
+
+    public function __construct(FactoryInterface $menuFactory, RouterInterface $router, ?TranslatorInterface $translator, Security $security, Connection $connection)
+    {
+        parent::__construct($menuFactory, $router, $translator);
+
+        $this->security = $security;
+        $this->connection = $connection;
+    }
+
+    /**
+     * {@inheritdoc}
+     */
+    public function supportsContext($context): bool
+    {
+        return 'content' === $context && $this->security->isGranted('contao_user.modules', $this->getBackendModule());
+    }
+
+    /**
+     * {@inheritdoc}
+     */
+    public function supportsValue(PickerConfig $config): bool
+    {
+        if (!is_numeric($config->getValue())) {
+            return false;
+        }
+
+        $qb = $this->connection->createQueryBuilder();
+        $qb
+            ->select('ptable')
+            ->from('tl_content')
+            ->where($qb->expr()->eq('id', $config->getValue()))
+        ;
+
+        if ($qb->execute()->fetchColumn() === $this->getParentTable()) {
```

Comment:
This can be simplified to `return $qb->execute()->fetchColumn() === $this->getParentTable()`, can't it?

Reply:
Changed in b1eeebb30a3fa26f5bc1c042e91ed6889585acb1